substrate/frame/support/test/tests/instance.rs

```diff
@@ -441,6 +441,7 @@ fn expected_metadata() -> PalletStorageMetadataIR {
 				ty: StorageEntryTypeIR::Plain(scale_info::meta_type::<u32>()),
 				default: vec![0, 0, 0, 0],
 				docs: vec![],
+				deprecation_info: sp_metadata_ir::DeprecationStatus::NotDeprecated,
```

Comment:
I would add a few tests similar to these (with/without note=.. since=..):
- check deprecation info for storage entries
- check deprecation info for runtime APIs for individual methods and for the trait itself
- check deprecation info for constants 

This way we'll also have a good example and see the API in action from the user's perspective

Reply:
added more testcases for `StorageEntryMetadataIr` and constants metadata